utils/generic.py

```diff
@@ -1409,3 +1444,140 @@ def get_csq_from_struct(element: hl.expr.StructExpression, feature_type: str) ->
         )
 
     return hl.or_missing(hl.len(csq) > 0, csq)
+
+
+def get_array_element_type(array_expr: hl.expr.ArrayExpression) -> hl.HailType:
+    """
+    Returns the type of an array element.
+
+    :param ArrayExpression array_expr: The array expression to get the element type
+    :return: Hail type
+    :rtype: HailType
+    """
+    return array_expr.dtype.element_type
+
+
+def ht_to_vcf_mt(
+        info_ht: hl.Table,
+        pipe_delimited_annotations  : List[str] = INFO_VCF_AS_PIPE_DELIMITED_FIELDS
+ ) -> hl.MatrixTable:
+    """
+    Creates a MT ready for vcf export from a HT. In particular, the following conversions are done:
+    - All int64 are coerced to int32
+    - Fields specified by `pipe_delimited_annotations` will be converted from arrays to pipe-delimited strings
+
+    Note: The MT returned has no cols.
+
+    :param Table info_ht: Input HT
+    :param pipe_delimited_annotations: List of info fields (they must be fields of the ht.info Struct)
+    :return: MatrixTable ready for VCF export
+    :rtype: MatrixTable
+    """
+
+    def get_pipe_expr(array_expr: hl.expr.ArrayExpression) -> hl.expr.StringExpression:
+        return hl.delimit(array_expr.map(lambda x: hl.or_else(hl.str(x), "")), "|")
+
+    # Make sure the HT is keyed by locus, alleles
+    info_ht = info_ht.key_by('locus', 'alleles')
+
+    # Convert int64 fields to int32 (int64 isn't supported by VCF)
+    for f, ft in info_ht.info.dtype.items():
+        if ft == hl.dtype('int64'):
+            logger.warning(f"Coercing field info.{f} from int64 to int32 for VCF output. Value will be capped at int32 max value.")
+            info_ht = info_ht.annotate(
+                info=info_ht.info.annotate(**{f: hl.int32(hl.min(2**31 - 1, info_ht.info[f]))})
```

Comment:
well, this unfortunately means that you loose the information about any number greater than 2,147,483,647 ... It could potentially be bad for some info. For the current info fields, it's totally fine but an example of a situation where it could be bad would be it you were to store a global genomic position (so, max would be 3.2bln) in an info field. 
I'm open to suggestions for another behavior!

Reply:
ohh I see, thanks! it seems really silly that VCFs don't support int32 if this cap could impact some of the info values. the warning seems reasonable to me though.